engine/engine.gyp

```diff
@@ -22,6 +22,88 @@
 	'targets':
 	[
 		{
+			'target_name': 'extract_docs',
+			'type': 'none',
+			
+			'all_dependent_settings':
+			{
+				'variables':
+				{
+					'dist_aux_files':
+					[
+						# Gyp will only use a recursive xcopy on Windows if the path ends with '/'
+						'<(PRODUCT_DIR)/extracted_docs/',
+					],
+				},
+			},
+			
+			'variables':
+			{
+				'conditions':
+				[
+					[
+						'host_os == "linux"',
+						{
+							'engine': '<(PRODUCT_DIR)/server-community',
+						},
+					],
+					[
+						'host_os == "mac"',
+						{
+							'engine': '<(PRODUCT_DIR)/server-community',
+						},
+					],
+					[
+						'host_os == "win"',
+						{
+							'engine': '<(PRODUCT_DIR)/server-community.exe',
+						},
+					],
+				],
+			},
+			
+			'dependencies':
+			[
+				# Requires a working LiveCode engine
+				'server',
+			],
+			
+			'sources':
+			[
+				'../extensions/script-libraries/oauth2/oauth2.livecodescript',
+			],
+			
+			'actions':
+			[
+				{
+					'action_name': 'extract_docs_from_stacks',
+					'message': 'Extracting docs from stacks',
+					
+					'inputs':
+					[
+						'../util/extract-docs.livecodescript',
+						'../ide-support/revdocsparser.livecodescript',
+						'<@(_sources)',
+					],
```

Comment:
I think you need to list `'<(engine)'` in the `inputs` so that extracting the documentation gets ordered after compiling the engine by the build process.